tests/MongoDB.Driver.Examples/DocumentationExamples.cs

```diff
@@ -1246,6 +1248,332 @@ public void Example_58()
             Render(filter).Should().Be("{ status : \"D\" }");
         }
 
+        [Fact]
+        public void Aggregation_Example_1()
+        {
+            RequireServer.Check().Supports(Feature.Aggregate);
+
+            //db.sales.aggregate([ 
+            //    { $match : { "items.fruit":"banana" } },
+            //    { $sort : { "date" : 1 } }
+            //])
+
+            // Start Aggregation Example 1
+            var aggregatePipeline = new EmptyPipelineDefinition<BsonDocument>()
+                .Match(Builders<BsonDocument>.Filter.Eq("items.fruit", "banana"))
+                .Sort(Builders<BsonDocument>.Sort.Ascending("date"));
+
+            var cursor = collection.Aggregate(aggregatePipeline);
+            // End Aggregation Example 1
+
+            Render(aggregatePipeline).Should().Be("[{ $match : { \"items.fruit\" : \"banana\" } },   { $sort : { \"date\" : 1 } }]");
+        }
+
+        [Fact]
+        public void Aggregation_Example_2()
+        {
+            RequireServer.Check().Supports(Feature.Aggregate);
+
+            //db.sales.aggregate([
+            //{
+            //    $unwind: "$items"
+            //},
+            //{   $match: { "items.fruit" : "banana", }
+            //},
+            //{
+            //    $group: { _id: { day: { $dayOfWeek: "$date" } }, count: { $sum: "$items.quantity" } }
+            //},
+            //{
+            //    $project: { dayOfWeek: "$_id.day", numberSold: "$count", _id: 0 }
+            //},
+            //{
+            //    $sort: { "numberSold": 1 }
+            //}])
+
+            // Start Aggregation Example 2
+            var aggregatePipeline = new EmptyPipelineDefinition<BsonDocument>()
+                .Unwind("items")
+                .Match(Builders<BsonDocument>.Filter.Eq("items.fruit", "banana"))
+                .Group(
+                    document => new BsonDocument("day", new BsonDocument("$dayOfWeek", "$date")),
```

Comment:
As I understand there is no big difference between POCO and BsonDocument in this context.
When you use Expression with POCO to specify _id value, you set the POCO property which will be used as _id in Translate logic.

In case if you use BsonDocument instead POCO in expression, you also specify _id key, but the field property you get not from POCO object, but from new BsonDocument. Then, this bson document will be Translated to ConstantExpression and then to BsonValue which will be used as idProjection.

But I agree, my implementation has a style disadvantage. I don't use the provided expression parameter (document). Maybe it's better to escape this.